src/platform_impl/android/mod.rs

```diff
@@ -160,50 +203,66 @@ impl<T: 'static> EventLoop<T> {
                 Some(EventSource::User) => {
                     let mut user_queue = self.user_queue.lock().unwrap();
                     while let Some(event) = user_queue.pop_front() {
-                        event_handler(
-                            event::Event::UserEvent(event),
+                        call_event_handler!(
+                            event_handler,
                             self.window_target(),
-                            &mut cf,
+                            control_flow,
+                            event::Event::UserEvent(event)
                         );
                     }
                 }
-                None => {}
+                None => {
+                    control_flow = ControlFlow::Exit;
```

Comment:
This causes my current application to restart in a loop.
Not sure what's the intention here as `first_event` is initialized as `None` on `EventLoop` creation which causes this match to always end up in the branch imo

Reply:
Are you using `run_return` directly or `return`?

While the control flow is set to exit there the first time it is not necessarily permanent. There is a chance for the user to change it later to something else when `MainEventsCleared` or `RedrawRequested` is emitted. Than the first event is set to something other than none. Also the next time we enter the loop `first_event` won't (unless the poll returns None) be None anymore because it is assigned to something else down below.

Reply:
Ah I see why this would be problematic when using `run`.

Reply:
This works with the winit `window` example which uses `run` because the control flow is always set to ` *control_flow = ControlFlow::Wait;` by the user.

Reply:
> Are you using run_return directly or return

`run`

> While the control flow is set to exit there the first time it is not necessarily permanent. There is a chance for the user to change it later to something else when MainEventsCleared or RedrawRequested is emitted.

That's not the case as far as I can see. In particular, that's caused by the way `call_event_handler` macro is implemented (https://github.com/VZout/winit/blob/6ce0fa2f09f94ecfad6ac3c37732c6c1f39f2401/src/platform_impl/android/mod.rs#L58). It creates a local variable which will throw away the value we set the control flow in our event handler.

> This works with the winit window example which uses run because the control flow is always set to *control_flow = ControlFlow::Wait; by the user.

Maybe, my phone works different, but I also overwrite it to `ControlFlow::Wait` at the beginning.

Reply:
Hm your right. The call_event_handler would prevent that. I remember testing this with `run` and it working but maybe I messed something up.